swagger.yaml

```diff
@@ -146,226 +328,800 @@ paths:
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/DatasetResponse'
+      limit:
+        description: "The number of datasets requested"
+        type: integer
+      offset:
+        description: "The first row of datasets to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+        type: integer
+      total_count:
+        description: "The total number of datasets"
+        readOnly: true
+        type: integer
+  DatasetResponse:
+    description: "A model for the response body when getting a dataset"
+    allOf:
+    - type: object
+      properties:
+        id:
+          readOnly: true
+          type: string
+          description: "An unique id for a dataset"
+          example: "DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC"
+    - $ref: "#/definitions/Dataset"
   Dataset:
+    description: "The dataset"
     type: object
     properties:
-      id:
-        type: string
-        description: "An unique id for a dataset"
-        example: "DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC"
-      title:
-        type: string
-        description: "The title of the dataset"
-        example: "CPI"
+      contact:
+        type: array
+        description: "Object containing contact information for this dataset"
+        items:
+          $ref: '#/definitions/Contact'
       description:
         type: string
         description: "A description for a dataset"
-      theme:
-        type: string
-        description: "The release date of this dataset"
+      links:
+        $ref: '#/definitions/DatasetLinks'
       next_release:
         type: string
         description: "The next release date for this dataset"
-      edition:
-        description: "A theme for a dataset"
       periodicity:
         type: string
         description: "The periodicity for a dataset"
       publisher:
-        description: "The publisher of the dataset"
         $ref: '#/definitions/Publisher'
-      latest_url:
+      state:
+        $ref: '#/definitions/State'
+      theme:
         type: string
-        description: "A link to the latest version and edition of the dataset"
-      contact:
-        type: array
-        description: "Object containing contact information for this dataset"
-        items:
-          $ref: '#/definitions/Contact'
-      editions_url:
+        description: "A theme for a dataset"
+      title:
         type: string
-        description: "A URL to all editions for a dataset"
-        example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/editions"
-  Datasets:
+        description: "The title of the dataset"
+        example: "CPI"
+  DatasetLinks:
+    description: "A list of links related to this resource"
+    readOnly: true
     type: object
     properties:
-      items:
-        type: array
-        description: "A list of datasets"
-        items:
-          $ref: '#/definitions/Dataset'
-  Edition:
+      editions:
```

Comment:
links should all be nested as objects for consistency

Reply:
The DatasetLinks model is nested in a links properties in the Dataset model